stdlib/public/runtime/Casting.cpp

```diff
@@ -169,7 +169,9 @@ void
 swift::swift_dynamicCastFailure(const void *sourceType, const char *sourceName, 
                                 const void *targetType, const char *targetName, 
                                 const char *message) {
+#if !defined(_WIN32)
   asm("");
```

Comment:
The intent is to preserve that frame in stack traces. I know the asm("") is necessary when the function is empty. This function is not empty so noinline alone may be sufficient.